### PR TITLE
Fix bad import of AUTH_USER_MODEL

### PR DESCRIPTION
**Pull request by marcusmartins**
The current mechanism used to load the user model in our fork of waffle
causes a lot of issues with circular imports. The fix here addresses
the issue by not assuming that django would be ready by the time that
waffle is imported.

Our version of waffle is pretty ancient but the API has changed so it
would require refactoring of our code base to bring it up to date.

cc @nwt @jmwong @Syntaxide

### CODE REVIEW

**Review by nwt — APPROVED** [review]
:shipit:

File: waffle/compat.py [hunk 1]
```diff
@@ -1,13 +1,3 @@
...
-# Django 1.5+ compatibility
-if django.VERSION >= (1, 5):
-    from django.contrib.auth import get_user_model
-    from django.conf import settings
-    User = get_user_model()
-    AUTH_USER_MODEL = settings.AUTH_USER_MODEL
-else:
-    from django.contrib.auth.models import User
-    AUTH_USER_MODEL = 'auth.User'
+AUTH_USER_MODEL = getattr(settings, 'AUTH_USER_MODEL', 'auth.User')
```
Comment (Syntaxide) [inline waffle/compat.py:3]: How does the default work here? I'm surprised that AUTH_USER_MODEL could either be a class or a string.

Comment (marcusmartins) [reply waffle/compat.py:3]: It's always a string. settings.AUTH_USER_MODEL should reference to the module to be imported.

https://docs.djangoproject.com/en/1.10/topics/auth/customizing/#substituting-a-custom-user-model

**Comment by marcusmartins** [comment]
I am merging this now to unblock tutum.

**Comment by jmwong** [comment]
+1

**Pull request merged**